app/models/assignment.rb

```diff
@@ -851,7 +853,59 @@ def get_num_valid
              .count(&:is_valid?)
   end
 
-  def get_num_marked(ta_id = nil)
+  def marked_result_ids_for(ta_id)
+    cache_ta_results.dig(ta_id, :marked_result_ids) || []
+  end
+
+  def cache_ta_results
+    return @cache_ta_results if defined? @cache_ta_results
+
+    data = current_results.joins(grouping: :tas).pluck('tas.id', 'results.id', 'results.marking_state')
+    # Group results by TA ID
+    grouped_data = data.group_by { |ta_id, _result_id, _marking_state| ta_id }
+    current_results.includes(:marks).index_by(&:id)
```

Comment:
@hemmatio I'm not sure what the purpose of this line is any more. It looks like you just deleted the variable (maybe because of a rubocop issue?) but in general if a variable is unused, you should almost always deleting the entire assignment statement (unless there are side effects; I don't think that applies here).